ballista/executor/src/replicator/mod.rs

```diff
@@ -148,37 +149,89 @@ fn serialize_batch(batch: RecordBatch) -> Result<bytes::Bytes, BallistaError> {
     Ok(Bytes::from(serialized_data))
 }
 
-async fn replicate_to_object_store(
-    destination: &Path,
+async fn upload_to_object_store(
+    mut writer: Box<dyn AsyncWrite + Send + Unpin>,
     mut reader: AsyncStreamReader<BufReader<Compat<File>>>,
-    object_store: Arc<dyn ObjectStore>,
 ) -> Result<(), BallistaError> {
-    let (_, mut upload) = object_store.put_multipart(destination).await.map_err(|e| {
-        REPLICATION_FAILURE
-            .with_label_values(&["create_writer"])
-            .inc();
-        BallistaError::General(format!("Failed to create object store writer: {:?}", e))
-    })?;
+    let written = std::sync::atomic::AtomicU64::new(0);
```

Comment:
This is just a local variable so you shouldn't need an atomic here. You can just use a `let mut written = 0;`

Reply:
yes, you're right

Reply:
Removed